src/linter.ts

```diff
@@ -8,13 +8,14 @@ import resolve from 'resolve'
 
 const prettierCache = new LRU<CacheInstance>(10)
 
-function createCache(cwd: string) {
-  let prettierPath
-  try {
-    prettierPath = resolve.sync('prettier', { basedir: cwd })
-  } catch (e) {
-    // module not found
-    prettierPath = resolve.sync('prettier')
+function createCache(cacheKey: string, cwd: string, prettierPath?: string) {
+  if (prettierPath == null) {
```

Comment:
Hmm, when the path is provided, we should probably run resolve with the cwd to support relative paths. I think this currently only works with fully resolved paths right now.

Reply:
Hm, I'm not totally worried about that, I'm thinking if a person is using `--prettier-path`, they can easily supply a full path (via `--prettier-path="${PWD}/.yarn/sdks/"`.

Is there no `yarn sdk prettier` command that dumps out a full path? I don't much use yarn in personal projects, and at work we use yarn v1, and the two versions can't coexist.

If anything, we could add an `else` below, and only resolve with the cwd if the given `prettierPath` doesn't start with a `/`.

Reply:
That's fair. It's easy enough to add later and a backwards compatible change if the need arises.